### ISSUE TITLE
Feature Request: Provide an option to specify the fallback shell type

### ISSUE THREAD

**Issue by moonfruit** [koalaman/shellcheck#2971]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Description

Some tools like [bash-language-server](https://github.com/bash-lsp/bash-language-server) detect the shell type themselves and add the option `--shell=SHELLNAME` to `shellcheck`.
But this option will overwrite `shellcheck` directive for type in file or `.shellcheckrc`.

I think a new option should be provided to specify a fallback shell type.

**Comment by brother** [comment]
Not sure this is a shellcheck problem really.

Would it be possible to add the [shell directive](https://github.com/koalaman/shellcheck/wiki/Directive#shell) at the top of the file?

```shell
brother ~$ cat /tmp/test.sh 
#!/bin/sh

myvar=(1 2 3)

echo "${myvar[1]}"
brother ~$ shellcheck --norc /tmp/test.sh 

In /tmp/test.sh line 3:
myvar=(1 2 3)
      ^-----^ SC3030 (warning): In POSIX sh, arrays are undefined.


In /tmp/test.sh line 5:
echo "${myvar[1]}"
      ^---------^ SC3054 (warning): In POSIX sh, array references are undefined.

For more information:
  https://www.shellcheck.net/wiki/SC3030 -- In POSIX sh, arrays are undefined.
  https://www.shellcheck.net/wiki/SC3054 -- In POSIX sh, array references are...
brother ~$ shellcheck --norc --shell=bash /tmp/test.sh 
brother ~$ cat /tmp/test.sh 
#!/bin/sh
# shellcheck shell=bash

myvar=(1 2 3)

echo "${myvar[1]}"
brother ~$ shellcheck --norc --shell=bash /tmp/test.sh 
brother ~$ shellcheck --norc /tmp/test.sh 
```

**Comment by moonfruit** [comment]
For example:

```console
$ cat test.sh
#!/bin/sh
# shellcheck shell=bash
echo -e "abc"
$ shellcheck test.sh
$ shellcheck --shell=sh test.sh

In test.sh line 3:
echo -e "abc"
     ^-- SC3037 (warning): In POSIX sh, echo flags are undefined.

For more information:
  https://www.shellcheck.net/wiki/SC3037 -- In POSIX sh, echo flags are undef...
```

**Comment by moonfruit** [comment]
[bash-language-server](https://github.com/bash-lsp/bash-language-server) always sets `--shell=` to what it thinks the shell is.

**Comment by brother** [comment]
maybe that should stop doing such things then?

**Comment by moonfruit** [comment]
Well, now that you've said that, all I can do now is post this question to bash-language-server.

**Comment by moonfruit** [comment]
It seems to me that a small change on both sides would be the feature that still works. Otherwise, an external tool may be needed to parse the `shellcheck` directive to solve the problem more perfectly.

**Comment by brother** [comment]
I don't think this need to get all crazy but it looks like bash-lang-server is (lack of words) abusing the shell directive. We developers and and end consumers can set those options ourselves if needed.

**Comment by ale5000-git** [comment]
@moonfruit
shellcheck already detect the shell by itslef in multiple ways so there isn't any need for the server to suggest it (even more so if the suggestions are wrong).

**Comment by moonfruit** [comment]
I'm not the maintainer of `bash-language-server`, I can't decide what it will do.
I just made a suggestion.
